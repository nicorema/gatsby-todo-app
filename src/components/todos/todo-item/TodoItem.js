import React from 'react'
import DeleteIcon from '../../../images/icons/DeleteIcon'
import EditIcon from '../../../images/icons/EditIcon'
import Card from '../../common/card/Card'
import { StyledItemContainer } from './TodoItem.styles'

const TodoItem = ({ deleteTodo, editTodo, todo }) => {
  return (
    <Card>
      <StyledItemContainer>
        <p>{todo?.description}</p>
        <div>
          <span
            role="button"
            onClick={() => deleteTodo(todo.id)}
            onKeyDown={() => deleteTodo(todo.id)}
            tabIndex={0}
            data-testid="delete-todo"
          >
            <DeleteIcon />
          </span>
          <span
            role="button"
            onClick={() => editTodo(todo)}
            onKeyDown={() => editTodo(todo)}
            tabIndex={0}
            data-testid="edit-todo"
          >
            <EditIcon />
          </span>
        </div>
      </StyledItemContainer>
    </Card>
  )
}

export default TodoItem
