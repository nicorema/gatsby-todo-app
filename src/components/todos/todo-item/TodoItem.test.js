import React from 'react'
import renderer from 'react-test-renderer'

import TodoItem from './TodoItem'

describe('TodoItem', () => {
  const mockDeleteTodo = jest.fn()
  const mockEditTodo = jest.fn()

  const todoItemCmp = renderer.create(
    <TodoItem
      deleteTodo={mockDeleteTodo}
      editTodo={mockEditTodo}
      todo={{ id: 0, description: 'test todo' }}
    />
  )

  it('renders correctly', () => {
    const snapShot = todoItemCmp.toJSON()
    expect(snapShot).toMatchSnapshot()
  })

  it('triggers deleteTodo', () => {
    const deleteBtn = todoItemCmp.root.findByProps({
      'data-testid': 'delete-todo',
    })
    deleteBtn.props.onClick()
    expect(mockDeleteTodo).toHaveBeenCalled()
  })

  it('triggers editTodo', () => {
    const editBtn = todoItemCmp.root.findByProps({
      'data-testid': 'edit-todo',
    })
    editBtn.props.onClick()
    expect(mockEditTodo).toHaveBeenCalled()
  })
})
