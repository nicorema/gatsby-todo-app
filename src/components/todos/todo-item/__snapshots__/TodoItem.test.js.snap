// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem renders correctly 1`] = `
<div
  className="css-fmx4z3"
>
  <div
    className="sc-bdnylx ikUfnl"
  >
    <p>
      test todo
    </p>
    <div>
      <span
        data-testid="delete-todo"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <svg
          height="14pt"
          viewBox="0 0 14 14"
          width="14pt"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 .555L9.445 0 5 4.445.555 0 0 .555 4.445 5 0 9.445.555 10 5 5.555 9.445 10 10 9.445 5.555 5zm0 0"
          />
        </svg>
      </span>
      <span
        data-testid="edit-todo"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <svg
          height="14pt"
          viewBox="0 0 14 14"
          width="14pt"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M8.215 5c-.2 0-.36.16-.36.355V8.93c0 .195-.16.355-.355.355H1.07a.357.357 0 01-.355-.355V1.785c0-.195.16-.355.355-.355h4.285a.358.358 0 100-.715H1.07C.48.715 0 1.195 0 1.785V8.93C0 9.52.48 10 1.07 10H7.5c.59 0 1.07-.48 1.07-1.07V5.355A.357.357 0 008.215 5zm0 0"
            />
            <path
              d="M9.602.402a1.369 1.369 0 00-1.938 0L2.961 5.105a.37.37 0 00-.086.141l-.715 2.14c-.062.188.04.391.227.454a.361.361 0 00.113.015c.04 0 .078-.003.113-.015l2.14-.715a.37.37 0 00.142-.086l4.703-4.703A1.369 1.369 0 009.602.402zm0 0"
            />
          </g>
        </svg>
      </span>
    </div>
  </div>
</div>
`;
